### PR TITLE
drivers: introduce nxp LCDIC MIPI DBI driver

### PR DESCRIPTION
**Pull request by danieldegrasse**
Introduce driver the NXP LCDIC, using the MIPI DBI driver class. This peripheral supports SPI 3/4 wire mode and 8080 mode, although only 4 wire mode is tested within the driver.

The LCDIC driver has been tested and enabled on the RW612 BGA board, with the Adafruit TFT Touch v2 shield.

~Note that this PR depends on the following PRs, and will be kept in draft until they merge:~
~- https://github.com/zephyrproject-rtos/zephyr/pull/70192~
~- https://github.com/zephyrproject-rtos/zephyr/pull/70193~
~- https://github.com/zephyrproject-rtos/zephyr/pull/70190~

### CODE REVIEW

File: drivers/mipi_dbi/mipi_dbi_nxp_lcdic.c [hunk 1]
```diff
@@ -0,0 +1,803 @@
...
+
+/* After LCDIC is enabled or disabled, there should be a wait longer than
+ * 5x the module clock before other registers are read
+ */
+static inline void mipi_dbi_lcdic_reset_delay(void)
+{
+	/* With a core clock of 260MHz, we need a minimum wait of 13
+	 * cycles to satisfy requirement. Wait for 130 as this is
+	 * the same value MCUX SDK uses.
+	 */
```
Comment (mmahadevan108) [inline drivers/mipi_dbi/mipi_dbi_nxp_lcdic.c:140]: Given this driver can be reused on other SoC's with varying core speeds, is it safe to have a loop like the below. Should we instead use Zephyr delay functions?

Comment (danieldegrasse) [reply drivers/mipi_dbi/mipi_dbi_nxp_lcdic.c:140]: No- thanks for catching this. I'll switch this to use `k_busy_wait`

**Comment by danieldegrasse** [comment]
Rebased to clear merge conflicts

**Review by decsny — DISMISSED** [review]

**Review by dleach02 — DISMISSED** [review]

**Comment by danieldegrasse** [comment]
Rebased to clear merge conflicts

**Review by mmahadevan108 — APPROVED** [review]

**Review by decsny — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Pull request merged**